bench/runbench.py

```diff
@@ -0,0 +1,167 @@
+    if jit:
+        rvm['id'] = 'rho-jit'
+    else:
+        rvm['id'] = 'rho'
+    try:
+        os.chdir(args.build_dir)
+        if not os.path.isdir('.git'):
+            # Clone Rho into the local directory.
+            exit_code = subprocess.call(['git', 'clone', args.repository, '.'])
+        # Clean and switch to the selected revision.
+        subprocess.call(['git', 'reset', '--hard', args.gitref])
+        # Build with JIT enabled.
+        subprocess.call(['git', 'clean', '-fd', 'HEAD'])
+        subprocess.call(['git', 'clean', '-fX', 'HEAD'])
+        if build:
+            subprocess.call(['tools/rsync-recommended'])
+            if jit:
+                subprocess.call(['./configure', '--with-blas', '--with-lapack', '--enable-llvm-jit'])
+            else:
+                subprocess.call(['./configure', '--with-blas', '--with-lapack'])
+            subprocess.call(['make', '-j2'])
+    finally:
+        os.chdir(bench_dir)
+    # Create a new config file because the benchmark suite is not aware of our
+    # Rho build.
+    rvm['cfg_file'] = write_config(args, rvm, jit)
+    return rvm
+
+# Write custom config file for running Rho in the benchmark suite.
+def write_config(args, rvm, jit):
+    config = ConfigParser.RawConfigParser()
+    config.add_section('GENERAL')
+    config.set('GENERAL', 'WARMUP_REP', 2)
+    config.set('GENERAL', 'BENCH_REP', 5)
+    config.set('GENERAL', 'PERF_TMP', '_perf.tmp')
+    config.set('GENERAL', 'PERF_REP', 1)
+    config.set('GENERAL', 'PERF_CMD', 'perf stat -r %(PERF_REP)s -x, -o %(PERF_TMP)s --append')
+    config.add_section('Rho')
+    if jit:
+        config.set('Rho', 'ENV', 'R_COMPILE_PKGS=1 R_ENABLE_JIT=2')
+    else:
+        config.set('Rho', 'ENV', 'R_COMPILE_PKGS=0 R_ENABLE_JIT=0')
+    config.set('Rho', 'HOME', os.path.realpath('%s/bin' % args.build_dir))
+    config.set('Rho', 'CMD', 'Rscript')
+    config.set('Rho', 'ARGS', '--vanilla')
+    config.set('Rho', 'HARNESS', 'r_harness.R')
+    config.set('Rho', 'HARNESS_ARGS', 'FALSE')
+    cfg_file = normpath('%s/rho.cfg' % args.result_dir)
+    with open(cfg_file, 'wb') as f:
+        config.write(f)
+    return cfg_file
+
+def bench(args, rvm):
+    for bm in benchmarks:
+        bench_cmd = [
+                'python', normpath('benchmarks/utility/rbench.py'),
+                normpath(bm['name']),
+                '--rvm', rvm['name'],
+                '--warmup_rep', "%d" % bm['warmup_rep'],
+                '--bench_rep', "%d" % bm['bench_rep'],
+                '--timingfile',
+                normpath('%s/%s-%s.csv' % (args.result_dir, rvm['id'], args.gitref))]
+        if 'cfg_file' in rvm:
+            bench_cmd = bench_cmd + ['--config', rvm['cfg_file']]
+        subprocess.call(bench_cmd)
+
+def main():
+    args = parse_args()
+    if not os.path.isdir('benchmarks'):
+        # Clone the benchmark suite.
+        subprocess.call(['git', 'clone', 'git@github.com:llbit/rbenchmarks.git', 'benchmarks'])
```

Comment:
I needed a few extra commits that were not merged in to the original repository, hence the fork. I will make PRs for the new repo.